### PR TITLE
Site Editor: Avoid using edited entity state in site editor loading hook

### PR DESCRIPTION
**Pull request by youknowriad**
Related #66921 

## What?

As the site editor is growing to become a multi page app, it doesn't really make sense to have an edited post type and context state in it.
That state is forcing us today to have a two synchronization mechanism between the url and the state causing some issues related to performance and some hidden bugs at times.

Ideally the routes would just pass the post type, post id to the editor component. The current PR helps to go in that direction by removing getEditedPostType usage from the isLoading hook.

## Testing Instructions

- Load the site editor, the loading should behave exactly like trunk.

### CODE REVIEW

File: packages/edit-site/src/components/layout/hooks.js [hunk 1]
```diff
@@ -5,15 +5,9 @@ import { useEffect, useState } from '@wordpress/element';
...
 
-/**
- * Internal dependencies
- */
-import useEditedEntityRecord from '../use-edited-entity-record';
-
 const MAX_LOADING_TIME = 10000; // 10 seconds
 
 export function useIsSiteEditorLoading() {
-	const { isLoaded: hasLoadedPost } = useEditedEntityRecord();
```
Comment (youknowriad) [inline packages/edit-site/src/components/layout/hooks.js:16]: my thinking is that this is a resolver as well, so there's no need to add this specifically here.

Comment (ntsekouras) [reply packages/edit-site/src/components/layout/hooks.js:16]: This makes sense. I added a conditional breakpoint though and sometimes `hasLoadedPost = false` and `loaded = true`.

Visually though I couldn't spot something different and I also tried with throttling the network speed.

Comment (youknowriad) [reply packages/edit-site/src/components/layout/hooks.js:16]: > This makes sense. I added a conditional breakpoint though and sometimes hasLoadedPost = false and loaded = true.

Was this during the initial loading or when you navigated between the pages? Seems the latter right and in that case it seems that it's probably ok.

Comment (ntsekouras) [reply packages/edit-site/src/components/layout/hooks.js:16]: The latter. Yeah, I think it will be fine.

**Review by ntsekouras — APPROVED** [review]
Thank you!

**Pull request merged**